tests/test_date.py

```diff
@@ -681,5 +681,16 @@ def then_date_object_is_invalid(self):
         self.assertFalse(self.is_valid_date_obj)
 
 
+class TestTimestampParser(BaseTestCase):
+    def setUp(self):
+        super(TestTimestampParser, self).setUp()
+
+    def test_timestamp_millis(self):
+        self.assertEqual(
+            date.get_date_from_timestamp(u'1570308760263', None),
+            datetime(2019, 10, 5, 23, 52, 40, 263)
+        )
```

Comment:
Your test is currently failing. I wonder if it has to do with the fact that you are passing milliseconds as the `microsecond` parameter in your implementation.

And you should also cover cases where the number has a different number of digits: 9, 11, 12, 14. I assume in those case the resulting date would be `None`. Maybe 16 digits (microseconds) should be supported as well.

Also, `setUp` can be removed from your test class.